### PR TITLE
Merge previous fixes back

### PR DESCRIPTION
**Pull request by sabina2014**
Removed sockets hashmap (https://issues.jboss.org/browse/JGRP-1420); 
Make members of MERGE2 thread safe.

### CODE REVIEW

File: build.xml [hunk 1]
```diff
@@ -6,7 +6,7 @@
         build.xml file for JGroups. Needs Ant (jakarta.apache.org) to run
     </description>
 
-    <property name="version" value="2.12.4.Final-TEK"/>
+    <property name="version" value="2.12.4.Final-TEK-1464-1420"/>
```
Comment (joninacjc) [inline build.xml:10]: Where does this 1464-1420 come from ?

File: src/org/jgroups/JChannel.java [hunk 2]
```diff
@@ -2105,9 +2105,9 @@ private TimeScheduler getTimer() {
                     Map<String, Object> tmp_info=getInfo();
                     map.put("info", tmp_info != null? Util.mapToString(tmp_info) : "null");
                 }
-                if(key.equals("socks")) {
-                    map.put("socks", getOpenSockets());
-                }
+//                if(key.equals("socks")) {
```
Comment (joninacjc) [inline src/org/jgroups/JChannel.java]: Why comment  but not just remove the code ?

File: src/org/jgroups/util/DefaultSocketFactory.java [hunk 3]
```diff
@@ -1,113 +1,94 @@
...
     }
 
     public void close(DatagramSocket sock) {
-        remove(sock);
         Util.close(sock);
     }
 
-    public Map<Object, String> getSockets() {
-        return sockets;
+    public Map<Object,String> getSockets() {
```
Comment (joninacjc) [inline src/org/jgroups/util/DefaultSocketFactory.java]: Is this method still used then ? Seems more dangerous to return null then just remove this method.

Comment (sabina2014) [inline src/org/jgroups/util/DefaultSocketFactory.java]: This method is called only by JChannel.MyProbeHandler.getOpenSockets() which is removed. 
I would remove it from DefaultSocketFactory and SocketFactory.

File: build.xml (hunk 1 above)
Comment (sabina2014) [inline build.xml:10]: Merged fix of JGRP1464 and JGRP1420. Don't know how to name the new version.

File: src/org/jgroups/JChannel.java (hunk 2 above)
Comment (sabina2014) [inline src/org/jgroups/JChannel.java]: Done.

File: build.xml (hunk 1 above)
Comment (joninacjc) [inline build.xml:10]: Ok that's fine. I would suggest add a comment to this pom file regarding the 2 fixes then.

Comment (sabina2014) [inline build.xml:10]: Done.

**Pull request merged**